cylc/flow/task_job_mgr.py

```diff
@@ -1222,14 +1223,21 @@ def _prep_submit_task_job(
 
         host_name, platform_name = None, None
         try:
-            if rtconfig['remote']['host'] is not None:
+            # We need to assume that we want a host if any of the items
+            # for the old host/batch system plaform selection system are set.
+            if any(
+                rtconfig[section][key[0]] is not None
+                for section, values in FORBIDDEN_WITH_PLATFORM.items()
```

Comment:
I understand checking `[job]batch system`, but I'm not sure whether `[host]retrieve job logs` should influence platform selection?

For reference:

```python
FORBIDDEN_WITH_PLATFORM: Dict[str, Dict[str, Set[Optional[str]]]] = {
    'remote': {
        # setting: exclusions
        'host': {'localhost', None},
        'retrieve job logs': {None},
        'retrieve job logs max size': {None},
        'retrieve job logs retry delays': {None},
    },
    'job': {
        'batch system': {None},
        'batch submit command template': {None}
    }
}
```

Reply:
If `[host]retrieve job logs` is set we can safely assume the workflow must not be using the Cylc 8 platforms syntax, as it would fail validation. So this seems fine to me